### DISCUSSION TITLE
Missing Actuator Groups in the recent changes on devel branch

### DISCUSSION THREAD

**Discussion by DexterFixxor** [isaac-sim/IsaacLab#153] [Q&A] [answered]
Hello, I could not help but notice that you removed support for the GroupActuators, like Robotiq gripper. Is there a specific reason for this?

The support for Grippers, especially the ones with mimic joints is crucial. It would be good If there was at least one example with Robotiq gripper given new API in devel branch.

**Comment by Mayankm96** [comment, marked as answer]
Hi,

Thanks a lot for bringing this up!

From the feedback we received, the previous design with actuator groups was confusing since it mixed two concepts:
1. ActuatorGroup: conversion of actions into joint commands (multiple-input multiple-output)
2. ActuatorModel: conversion of joint commands into simulator joint commands (single-input single-output)

The above made it very tempting to put control layers such as inverse kinematics as part of (1), which is not what we wanted it to be designed for. Thus, we changed the interfaces to be the following:
1. ActionManager: handles conversion of actions to joint commands
2. ActuatorModel: remains the same

This makes it easier to understand that all MIMO-related translations are in one place. However, it breaks the "concept" that all robot-related actuation constraints (mimicking, holonomic) should be bundled with the robot configuration. We had to make this trade-off for "better" scaling of design.

The action terms are now defined under the `orbit.envs.mdp.actions` sub-module: https://github.com/NVIDIA-Omniverse/Orbit/tree/devel/source/extensions/omni.isaac.orbit/omni/isaac/orbit/envs/mdp/actions

An example of using this landed in one of our commits today that shows how to use the gripper action for Franka robot: 1b293e155b03ad46d7b7815c5e636c57b48d2d9d

I hope this helps answer the question!
